src/strands/experimental/bidirectional_streaming/types/bidirectional_streaming.py

```diff
@@ -6,231 +6,483 @@
+        return cast(str, self.get("session_id"))
 
-    Signals when user interruption is detected during model generation.
+    @property
+    def model(self) -> str:
+        return cast(str, self.get("model"))
 
-    Attributes:
-        reason: Interruption reason from predefined set.
+    @property
+    def capabilities(self) -> List[str]:
+        return cast(List[str], self.get("capabilities"))
+
+
+class TurnStartEvent(TypedEvent):
+    """Model starts generating a response.
+
+    Parameters:
+        turn_id: Unique identifier for this turn (used in turn.complete).
     """
 
-    reason: Literal["user_input", "vad_detected", "manual"]
+    def __init__(self, turn_id: str):
+        super().__init__({"type": "bidirectional_turn_start", "turn_id": turn_id})
 
+    @property
+    def turn_id(self) -> str:
+        return cast(str, self.get("turn_id"))
 
-class BidirectionalConnectionStartEvent(TypedDict, total=False):
-    """connection start event for bidirectional streaming.
 
-    Attributes:
-        connectionId: Unique connection identifier.
-        metadata: Provider-specific connection metadata.
+class AudioStreamEvent(TypedEvent):
+    """Streaming audio output from the model.
+
+    Parameters:
+        audio: Base64-encoded audio string.
+        format: Audio encoding format.
+        sample_rate: Number of audio samples per second in Hz.
+        channels: Number of audio channels (1=mono, 2=stereo).
     """
 
-    connectionId: Optional[str]
-    metadata: Optional[Dict[str, Any]]
+    def __init__(
+        self,
+        audio: str,
+        format: Literal["pcm", "wav", "opus", "mp3"],
+        sample_rate: Literal[16000, 24000, 48000],
+        channels: Literal[1, 2],
+    ):
+        super().__init__(
+            {
+                "type": "bidirectional_audio_stream",
+                "audio": audio,
+                "format": format,
+                "sample_rate": sample_rate,
+                "channels": channels,
+            }
+        )
+
+    @property
+    def audio(self) -> str:
+        return cast(str, self.get("audio"))
+
+    @property
+    def format(self) -> str:
+        return cast(str, self.get("format"))
+
+    @property
+    def sample_rate(self) -> int:
+        return cast(int, self.get("sample_rate"))
+
+    @property
+    def channels(self) -> int:
+        return cast(int, self.get("channels"))
+
+
+class TranscriptStreamEvent(TypedEvent):
+    """Audio transcription of speech (user or assistant).
+
+    Parameters:
+        text: Transcribed text from audio.
+        source: Who is speaking ("user" or "assistant").
+        is_final: Whether this is the final/complete transcript.
+    """
 
+    def __init__(
+        self, text: str, source: Literal["user", "assistant"], is_final: bool
+    ):
+        super().__init__(
+            {
+                "type": "bidirectional_transcript_stream",
+                "text": text,
+                "source": source,
```

Comment:
Why `source` and not `role`?  We use `role` in Message so that seems like it'd be more consistent